sensorium/training/trainers.py

```diff
@@ -68,14 +83,44 @@ def standard_trainer(
         min_lr: minimum learning rate
         cb: whether to execute callback function
         track_training: whether to track and print out the training progress
+
+        cluster_number: Give number of clusters for DEC clustering algortihm
+        alpha: alpha used for calculation of soft assignment
+        dec_starting_epoch: Epoch at which we start the initialisation for the cluster centroids for dec clustering
+        kmeans_init: number of iterations for kmeans for cluster initialisation
         **kwargs:
 
     Returns:
 
     """
 
-    def full_objective(model, dataloader, data_key, *args, **kwargs):
+    def get_multiplier(epoch, base_multiplier=1e8):
+        if epoch < 14:
+            return 0
+        else:
+            return base_multiplier
+
+    def soft_assignments(encoded_features, cluster_centers, alpha=alpha):
+        # Compute soft assingments q_ij as described in DEC paper (1)
```

Comment:
@Nisone2000 maybe add formula (1) here, such that a person who reads it does not have to go to the papers (also that you don't have to go to the paper in case we start debugging it)